### PR TITLE
Improve FTP and SIP fingerprinting of network cameras, etc

### PR DESCRIPTION
**Pull request by jhart-r7**
## Description

A variety of improvements that started with FTP in general and ended up with a few related cleanups.  

## Motivation and Context

More fingerprints for QNAP, Axis, improving Pure and Pro-ftpd fingerprints, Netgear ReadyNAS and more.  I also made Camera more consistent with the `Web camera` used elsewhere.  Eventually we'll standardize on one, but for now we stay consistent.  

## How Has This Been Tested?

New examples, GPRs.

## Types of changes
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have updated the documentation accordingly (or changes are not required).
- [ ] I have added tests to cover my changes (or new tests are not required).
- [ ] All new and existing tests passed.

### CODE REVIEW

File: xml/ftp_banners.xml [hunk 1]
```diff
@@ -201,6 +201,31 @@ example.com FTP server (Version:  Mac OS X Server) ready.</example>
...
   </fingerprint>
+  <fingerprint pattern="^ProFTPD (\d+\.[^\s]+) Server \(NETGEAR ReadyNAS\) \[(.+)\]$">
+    <description>ProFTPD on a Netgear ReadyNAS with a version and IP</description>
+    <example service.version="1.3.3g" host.ip="192.168.1.10">ProFTPD 1.3.3g Server (NETGEAR ReadyNAS) [192.168.1.10]</example>
+    <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
+    <param pos="0" name="service.product" value="ProFTPD"/>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:proftpd:proftpd:{service.version}"/>
+    <param hw="0" name="hw.vendor" value="Netgear"/>
```
Comment (tsellers-r7) [inline xml/ftp_banners.xml]: `hw=0` should be `pos=0` here and below.

File: xml/ftp_banners.xml [hunk 2]
```diff
@@ -201,6 +201,31 @@ example.com FTP server (Version:  Mac OS X Server) ready.</example>
...
+    <param hw="0" name="hw.product" value="ReadyNAS"/>
+    <param pos="2" name="host.ip"/>
+  </fingerprint>
+  <fingerprint pattern="^ProFTPD Server \(NETGEAR ReadyNAS\) \[(.+)\]$">
+    <description>ProFTPD on a Netgear ReadyNAS with a hostname</description>
+    <example host.name="test">ProFTPD Server (NETGEAR ReadyNAS) [test]</example>
+    <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
+    <param pos="0" name="service.product" value="ProFTPD"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:proftpd:proftpd:{service.version}"/>
```
Comment (tsellers-r7) [inline xml/ftp_banners.xml]: No `service.version` to be used w/ interpolation

File: xml/ftp_banners.xml [hunk 3]
```diff
@@ -282,18 +308,28 @@ more stuff</example>
...
 more text</example>
     <param pos="1" name="pureftpd.config"/>
     <param pos="0" name="service.family" value="Pure-FTPd"/>
     <param pos="0" name="service.product" value="Pure-FTPd"/>
   </fingerprint>
+  <fingerprint pattern="^(?:Welcome to )?Pure-FTPd\.?$">
+    <description>Basic Pure-FTPd banner, no version</description>
+    <example>Welcome to Pure-FTPd</example>
+    <example>Pure-FTPd.</example>
+    <param pos="1" name="pureftpd.config"/>
```
Comment (tsellers-r7) [inline xml/ftp_banners.xml]: Not captured/used.

File: xml/ftp_banners.xml [hunk 4]
```diff
@@ -511,25 +547,42 @@ more text</example>
...
     <example os.product="241Q">AXIS 241Q Video Server 4.47.2 (Dec 11 2008) ready.</example>
     <example os.version="5.07.2">AXIS P7701 Video Decoder 5.07.2 (Apr 20 2010) ready.</example>
+    <example hw.product="Q7401" os.version="5.01">AXIS Q7401 Video Encoder 5.01 (Aug 01 2008) ready.</example>
+    <example hw.product="Q7401" os.version="5.50.2_cst_412205_1">AXIS Q7401 Video Encoder 5.50.2_cst_412205_1 (2013)</example>
+    <example hw.product="Q7424-R" os.version="5.51.3.1">AXIS Q7424-R Mk II Video Encoder 5.51.3.1 (2016) ready.</example>
+    <example hw.product="P8221" os.version="5.10.2">AXIS P8221 IO Audio Module 5.10.2 (Nov 07 2011) ready.</example>
+    <param pos="0" name="hw.vendor" value="Axis"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="0" name="os.vendor" value="Axis"/>
+    <param pos="1" name="os.product"/>
```
[the remark this answers is not in the document]
Comment (jhart-r7) [reply xml/ftp_banners.xml]: yeah, agreed.  I copied the existing pattern but will likely change it for axis now.

**Pull request merged**